cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,180 @@
-        }
-        rest_vm = appliance.rest_api.collections.vms.get(name=vm)
-        if from_detail:
-            responses = [rest_vm.action.add_event(**event)]
-        else:
-            responses = appliance.rest_api.collections.vms.action.add_event(rest_vm, **event)
-        assert appliance.rest_api.response.status_code == 200
-        for response in responses:
-            assert response["success"] is True, "Could not add event"
-
-        events = appliance.db.client["event_streams"]
-        events_list = list(appliance.db.client.session.query(events).filter(
-            events.vm_name == vm,
-            events.message == event["event_message"],
-            events.event_type == event["event_type"],
-        ))
-        assert events_list, "Could not find the event in the database"
-
-    @pytest.mark.parametrize("from_detail", [True, False], ids=["from_detail", "from_collection"])
-    def test_vm_add_lifecycle_event(self, vm, from_detail, appliance):
-        """Test that checks whether adding a lifecycle event using the REST API works.
-        Prerequisities:
-            * A VM
-        Steps:
-            * Find the VM's id
-            * Prepare the lifecycle event data (``status``, ``message``, ``event``)
-            * Call either:
-                * POST /api/vms/<id> (method ``add_lifecycle_event``) <- the lifecycle data
-                * POST /api/vms (method ``add_lifecycle_event``) <- the lifecycle data
-                    and resources field specifying list of dicts containing hrefs to the VMs,
-                    in this case only one.
-            * Verify that appliance's database contains such entries in table ``lifecycle_events``
-        Metadata:
-            test_flag: rest
+        timelines_view = target.open_timelines()
+        timeline_filter = timelines_view.filter
+
+        for selected_option in timeline_filter.event_category.all_selected_options:
+            timeline_filter.event_category.select_by_visible_text(selected_option)
+
+        timeline_filter.event_category.select_by_visible_text(self.__dict__['tl_category'])
+        timeline_filter.time_position.select_by_visible_text('centered')
+        timeline_filter.apply.click()
+        events_list = timelines_view.chart.get_events(self.__dict__['tl_category'])
+        logger.debug('events_list: ', events_list)
+        logger.info('Searching for event type: %r in timeline category: %r', self.event,
+                    self.__dict__['tl_category'])
+
+        if not len(events_list):
+            self.vm.provider.refresh_provider_relationships()
+            logger.warn('Event list of %r is empty!', target)
+
+        found_events = []
+
+        for evt in events_list:
+            if hasattr(evt, 'destination_vm'):
+                #  Specially for the VmDeployedEvent where the source_vm defers from the
+                # self.vm.name
+                if evt.destination_vm == self.vm.name \
```

Comment:
REQUIRED: please use () instead of back slash if your statement doesn't fit to one line